### DISCUSSION TITLE
Companion App 2.0.19 not recognized on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by barefootlance** [aclap-dev/video-downloadhelper#184] [Help and Q&A]
### Your question

Successfully downloaded yesterday, but today something has gone awry. 

When I tried to download the video (https://vimeo.com/150423718 - old, but pretty funny stuff), I got a message that I needed to upgrade the companion app.

I uninstalled the companion app version I had (2.0.10), downloaded the current one (2.0.19), and installed it, but the addon no longer recognizes the helper as being installed. Firefox is what I usually use, but confirmed that the helper is "not installed" on Chrome as well.

Any idea what I might be doing wrong?

### Addon & CoApp version and details

Ubuntu 22.04 (all up-to-date)
Firefox 123.0 (not snap or flatpack, an old-fashioned manual install)

ffmpeg:
ffmpeg version 4.4.2-0ubuntu0.22.04.1 Copyright (c) 2000-2021 the FFmpeg developers
built with gcc 11 (Ubuntu 11.2.0-19ubuntu1)
configuration: --prefix=/usr --extra-version=0ubuntu0.22.04.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
libavutil      56. 70.100 / 56. 70.100
libavcodec     58.134.100 / 58.134.100
libavformat    58. 76.100 / 58. 76.100
libavdevice    58. 13.100 / 58. 13.100
libavfilter     7.110.100 /  7.110.100
libswscale      5.  9.100 /  5.  9.100
libswresample   3.  9.100 /  3.  9.100
libpostproc    55.  9.100 / 55.  9.100

ADDON:
version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: no.

Companion app (from /opt/vdhcoapp/vdhcoapp --info):
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/lance",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}

**Comment by dwbsovran** [comment]
I also have the exact same problem with Linux Mint 21.3.  Th Companion app is downloaded and installed.  I go to download video and EVERY time it asks me to install the Companion App.  When I download the Co-App it tells me the current version is already installed.

When I uninstall the Co-App and re-install, the same problem re-occurs.  It is a never ending loop.  I uninstall VDH and reinstall VDH and the Co-App and try to download a video, once again it tells me I need to install the Companion App even though it has been downloaded and installed.  A never ending loop of frustration!

**Reply by GrampaWildWilly** [reply]
I don't know squat about Linux.  But this is in our Wiki:

https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#ubuntu-and-flatpak--snap-based-systems

Is that of any use?

**Reply by paulrouget** [reply]
@dwbsovran can you give me more details:

- how do you install the coapp exactly
- after installed, when you go to the VDH settings, does it tell you that the coapp is recognized?
- does it tell you the coapp is outdated?

**Reply by Jackalaq** [reply]
SOLVED!
Ditto here on Linux Mint 21.2  and Firefox 120.0.1 (64 bit)

It requested an update of the CoApp, saying that a newer version was needed.  The installation reports that the same version is already installed and now the Settings page says that no app is installed. 

@GrampaWildWilly is totally correct.  That is the issue/solution.

**Ubuntu (and Flatpak / Snap based systems)**

First, make sure Ubuntu and Firefox are up to date. Firefox 122 on Ubuntu 22.04 had an issue that has been recently fixed.

Most of the time, the issue is a missed prompt. Firefox should have showed you a prompt to allow the coapp to communicate with the extension. Some people miss the prompt, or the prompt just never shows up. There's no user interface to change that setting. To solve this, run these commands (even if you don't use Flatpak):

/opt/vdhcoapp/vdhcoapp install _# (not as root! Don't use sudo)_
sudo apt-get install -y flatpak
flatpak permission-set webextensions net.downloadhelper.coapp snap.firefox yes


Thanks @GrampaWildWilly !

**Reply by gridsleep** [reply]
Many thanks, Jackalaq. As a famous loser once said, "I have always depended on the kindness of strangers."

**Comment by GrampaWildWilly** [comment]
I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 122.0.1 64-bit, licensed VDH 8.2.1.1a1 beta, CoApp 2.0.19.  And I don't work for VDH.  I'm just another user.

I went to your Vimeo video & it looked like this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f1ec365b-c9d7-47d8-a687-7a644a25cb9b)

A choice of 3 rather poor resolutions.  So I picked the top one & launched the download.  It looked like this on the VDH blue dot status menu:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c2702bbf-b586-4559-bba8-e348675f629e)

That is a frankly horrifically bad download speed.  I did watch it in the Resource Monitor for a while & it was fluctuating a lot.  This tells me the site is mercilessly throttling downloads.  In any case, it did finally complete successfully & I got this:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/9c6fd7e4-7b8f-46df-9c4a-76d053fe1aa8)

Using the numbers on the General tab gives me . . .

1,377,111,352 bytes/(60*47+32) seconds = 482,858.12 bytes per second

So that was its average download speed.  Pretty lousy.  Better than what I happened to capture on the VDH status menu, but still quite bad.

You will not be surprised to know that it played just fine in VLC.  I didn't sit & watch it, just skimmed through it sampling it here & there.  There was video & audio at every sample point, & the last 20 seconds also had video+audio.

Just to get a comparison, & because I thought I might put the time to good use while this thing was snailing along, I decided to try Plan B.

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/799a56b0-b49a-47f6-8185-06443d76d535)

Doing the usual trick of changing the .json in the URL of that master item to .m3u8 gave me this ffprobe report:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14353835/ffprobe.txt)

Same 3 lousy resolutions VDH offered.  Note that there is only 1 audio track shared by all 3 video tracks.  So I once again picked the highest one on offer (720x540) & downloaded that with ffmpeg:

[Oz Test mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14353838/Oz.Test.mp4.Log.txt)

You'll note in there that the speed factor ffmpeg was measuring started pretty low & got quite low.  But then it picked up & managed to end at a speed factor of 7.65, which is actually not bad.  As you can read in the log, this download averaged about 3 times the speed that VDH got.  I suspect some of the improvement was due to the time of day & the load on the server.  Maybe some users disconnected & left me some extra bandwidth.  But that's all speculation.  The thing to note is that the ffmpeg in the VDH CoApp is the ffmpeg I used to perform this download.  So it's hard to explain why VDH would be so much slower than my Plan B download.  For a time, both downloads were running concurrently.  Resource Monitor was showing me that Plan B was about 3 times faster than VDH.

This downloaded file looked like this:

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/d522f773-9c2c-437d-974d-5710212e1db1)

I can't explain the slight differences.  The files are not the same size.  My Plan B download is about 36 million bytes smaller.  That's significant.  The video total bitrate of the Plan B download is about 48kbps lower, & the audio bit rate for Plan B is 49kbps lower.  You'll also notice something that a number of people have been reporting.  The Plan B download is 4 seconds longer duration.  In this case, both files end after the music stops & the picture fades to black after the end credits, so you wouldn't even notice the truncation just watching the 2 videos.  But that's pure luck.  Other people have been reporting that significant content is being cut off.  It may be only a few seconds, but it isn't always just black padding that's being truncated.  In any case, the lower bit rates surely explain the smaller file size.

I suppose what I've proven is that if you can ever get your Linux to recognize the latest CoApp, you'll probably be able to download your video no problem.

**Comment by paulrouget** [comment]
I can reproduce the problem here. I'm investigating.

**Comment by paulrouget** [comment]
@barefootlance can you tell me which package you used exactly? .deb or tarball?

**Comment by dwbsovran** [comment]
Thanks Gramps!  Scroll to bottom - Linux (all) and just enter the two
commands and it now can find coapp.

Linux (all)

   - Run /opt/vdhcoapp/vdhcoapp install (*not as root! Don't use sudo*) to
   re-register the app.
   - Move to the installation folder of the coapp in your terminal, and
   run: /opt/vdhcoapp/vdhcoapp --info. It will run some diagnostic
   operations.
   - do not install vdhcoapp within the /usr/ directory, it won't be
   detected by your browser. See here why
   <https://github.com/aclap-dev/vdhcoapp/issues/160#issuecomment-1780765719>
   .


On Tue, Feb 20, 2024 at 9:09 PM GrampaWildWilly ***@***.***>
wrote:

> I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox
> 122.0.1 64-bit, licensed VDH 8.2.1.1a1 beta, CoApp 2.0.19. And I don't work
> for VDH. I'm just another user.
>
> I went to your Vimeo video & it looked like this:
>
> 01.png (view on web)
> <https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f1ec365b-c9d7-47d8-a687-7a644a25cb9b>
>
> A choice of 3 rather poor resolutions. So I picked the top one & launched
> the download. It looked like this on the VDH blue dot status menu:
>
> 02.png (view on web)
> <https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c2702bbf-b586-4559-bba8-e348675f629e>
>
> That is a frankly horrifically bad download speed. I did watch it in the
> Resource Monitor for a while & it was fluctuating a lot. This tells me the
> site is mercilessly throttling downloads. In any case, it did finally
> complete successfully & I got this:
>
> 03.png (view on web)
> <https://github.com/aclap-dev/video-downloadhelper/assets/75588964/9c6fd7e4-7b8f-46df-9c4a-76d053fe1aa8>
>
> Using the numbers on the General tab gives me . . .
>
> 1,377,111,352 bytes/(60*47+32) seconds = 482,858.12 bytes per second
>
> So that was its average download speed. Pretty lousy. Better than what I
> happened to capture on the VDH status menu, but still quite bad.
>
> You will not be surprised to know that it played just fine in VLC. I
> didn't sit & watch it, just skimmed through it sampling it here & there.
> There was video & audio at every sample point, & the last 20 seconds also
> had video+audio.
>
> Just to get a comparison, & because I thought I might put the time to good
> use while this thing was snailing along, I decided to try Plan B.
>
> 04.png (view on web)
> <https://github.com/aclap-dev/video-downloadhelper/assets/75588964/799a56b0-b49a-47f6-8185-06443d76d535>
>
> Doing the usual trick of changing the .json in the URL of that master item
> to .m3u8 gave me this ffprobe report:
>
> ffprobe.txt
> <https://github.com/aclap-dev/video-downloadhelper/files/14353835/ffprobe.txt>
>
> Same 3 lousy resolutions VDH offered. Note that there is only 1 audio
> track shared by all 3 video tracks. So I once again picked the highest one
> on offer (720x540) & downloaded that with ffmpeg:
>
> Oz Test mp4 Log.txt
> <https://github.com/aclap-dev/video-downloadhelper/files/14353838/Oz.Test.mp4.Log.txt>
>
> You'll note in there that the speed factor ffmpeg was measuring started
> pretty low & got quite low. But then it picked up & managed to end at a
> speed factor of 7.65, which is actually not bad. As you can read in the
> log, this download averaged about 3 times the speed that VDH got. I suspect
> some of the improvement was due to the time of day & the load on the
> server. Maybe some users disconnected & left me some extra bandwidth. But
> that's all speculation. The thing to note is that the ffmpeg in the VDH
> CoApp is the ffmpeg I used to perform this download. So it's hard to
> explain why VDH would be so much slower than my Plan B download. For a
> time, both downloads were running concurrently. Resource Monitor was
> showing me that Plan B was about 3 times faster than VDH.
>
> This downloaded file looked like this:
>
> 05.png (view on web)
> <https://github.com/aclap-dev/video-downloadhelper/assets/75588964/d522f773-9c2c-437d-974d-5710212e1db1>
>
> I can't explain the slight differences. The files are not the same size.
> My Plan B download is about 36 million bytes smaller. That's significant.
> The video total bitrate of the Plan B download is about 48kbps lower, & the
> audio bit rate for Plan B is 49kbps lower. You'll also notice something
> that a number of people have been reporting. The Plan B download is 4
> seconds longer duration. In this case, both files end after the music stops
> & the picture fades to black after the end credits, so you wouldn't even
> notice the truncation just watching the 2 videos. But that's pure luck.
> Other people have been reporting that significant content is being cut off.
> It may be only a few seconds, but it isn't always just black padding that's
> being truncated. In any case, the lower bit rates surely explain the
> smaller file size.
>
> I suppose what I've proven is that if you can ever get your Linux to
> recognize the latest CoApp, you'll probably be able to download your video
> no problem.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aclap-dev/video-downloadhelper/discussions/184#discussioncomment-8538287>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BGJ7ZVFZBSNFWXGBLUGFM7TYUVXOHAVCNFSM6AAAAABDR5DFRWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DKMZYGI4DO>
> .
> You are receiving this because you commented.Message ID:
> <aclap-dev/video-downloadhelper/repo-discussions/184/comments/8538287@
> github.com>
>


-- 
*Doug Barber*

**Comment by dwbsovran** [comment]
I installed VDH sometime ago from Firefox Extensions  Also installed the
Companion App.  Then on one of the VDH recent updates, Coapp stopped
working, giving me a notification to install the Companion App with a link
to GitHub.  In Linux Mint, I installed the 64 bit .deb package of CoApp..
That is the one that did not work or could not be recognized or found by
VDH.

The link provided by GrandpaWildWilly provided the FlatPack and Snap
installation which now works.

https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#ubuntu-and-flatpak--snap-based-systems

Just scroll to the bottom of the page to Linux (all) and type the two
commands into the terminal.  Presto!  It worked flawlessly.


On Tue, Feb 20, 2024 at 10:04 PM Paul ***@***.***> wrote:

> @dwbsovran <https://github.com/dwbsovran> can you give me more details:
>
>    - how do you install the coapp exactly
>    - after installed, when you go to the VDH settings, does it tell you
>    that the coapp is recognized?
>    - does it tell you the coapp is outdated?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aclap-dev/video-downloadhelper/discussions/184#discussioncomment-8538588>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BGJ7ZVGZLD3Z64UXCDLE6DDYUV56NAVCNFSM6AAAAABDR5DFRWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DKMZYGU4DQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <aclap-dev/video-downloadhelper/repo-discussions/184/comments/8538588@
> github.com>
>


-- 
*Doug Barber*
